src/search/data_loader.cpp

```diff
@@ -275,6 +275,12 @@ std::list<SearchEntity*> CDataLoader::GetPlayersList(search_req sr, int* count)
 
             PPlayer->flags2 = PPlayer->flags1;
 
+            // dont show anon in results if seraching by job, nation,  race, rank, lvl
+            if (FLAG_ANON && (sr.jobid > 0 || sr.nation != 255 || sr.race != 255 || sr.minRank > 0 || sr.maxRank > 0 || sr.minlvl > 0 || sr.maxlvl > 0))
```

Comment:
I suspect this needs to be:
`if ((nameflag & FLAG_ANON) && (sr.jobid > 0 || sr.nation != 255 || sr.race != 255 || sr.minRank > 0 || sr.maxRank > 0 || sr.minlvl > 0 || sr.maxlvl > 0))`
FLAG_ANON by itself is a constant and would always be true.

Reply:
good call, fixed